bsdlib/doc/gnss_extension.rst

```diff
@@ -143,11 +143,57 @@ Satellites with elevation angle less than the threshold are automatically exclud
 Use case
    | Socket option:  :c:type:`NRF_SO_GNSS_USE_CASE`
    | Datatype:       :c:type:`uint8_t`
-   | Default         1
-   | Allowed values: 0, 1
+   | Default         :c:type:`NRF_GNSS_USE_CASE_MULTIPLE_HOT_START | NRF_GNSS_USE_CASE_NORMAL_ACCURACY`
+   | Allowed values:
+
+   | :c:type:`NRF_GNSS_USE_CASE_SINGLE_COLD_START`
+   | :c:type:`NRF_GNSS_USE_CASE_MULTIPLE_HOT_START`
+   | :c:type:`NRF_GNSS_USE_CASE_NORMAL_ACCURACY`
+   | :c:type:`NRF_GNSS_USE_CASE_LOW_ACCURACY`
+
+This bitmask is used to specify the combination of the target GNSS performance (single cold or multiple hot start)
+and accuracy (normal or low).
+
+..Note::
+  Normal and Low accuracy is available in MFW 1.2.2 and later
```

Comment:
Perhaps write "nRF9160 modem firmware" or something like that instead of "MFW".

Reply:
Actually not sure if this line shall be included. here we need to follow how it's done for other bsdlib documentation where we have added feature that has been made available in a specific version of the MFW

Reply:
This is actually a problem in this document, but I'm a bit confused by NRF_GNSS_USE_CASE_SINGLE_COLD_START and NRF_GNSS_USE_CASE_NORMAL_ACCURACY. These are defined as "0 << 0" and "0 << 1" in nrf_socket.h, so both have a value of 0. Using these as normal bitmask values doesn't work, for example:

`if ((value & NRF_GNSS_USE_CASE_SINGLE_COLD_START) == NRF_GNSS_USE_CASE_SINGLE_COLD_START)`

This would always be true, regardless of the `value`. While setting the bits with these defines works, it's still confusing (at least to me).

Reply:
Had a look in existing BSD documentation and we haven't made such note earlier. We seems to referre to MFW change log for addd features. Then we shouldn't start doing it for this either. I.e. remove it.  (Only place where we could mention this will be Change log)

Reply:
@tokangas Isn't what you point's out here a problem in ".h" and/or ".c" and not as used in this documentation?
We have a pattern on how to define and use bit values in HW definition. Should we copy that for this?

Reply:
Those two zero bitmap defines are in nrf_socket.h:

[https://github.com/nrfconnect/sdk-nrfxlib/blob/master/bsdlib/include/nrf_socket.h#L357](https://github.com/nrfconnect/sdk-nrfxlib/blob/master/bsdlib/include/nrf_socket.h#L357)

This documentation should of course reflect what's defined in nrf_socket.h. I think we should only have defines in there which have the corresponding bit set. The defines which do not have any bit set are meaningless and can cause confusion.

I will discuss this with Even. I guess we can keep this as it is now because it reflects what's in nrf_socket.h. If nrf_socket.h is changed, this document needs to be also updated.

Reply:
Yes, I agree that thous defines is a bit strange, and I was not sure how to do them. The idea is that every setting should have a define, in this case a zero value in both bit 0 and bit 1 is a valid setting (not just turning something off) as documented in the rpc documentation.

Then you can write code lines like:

config.use_case = NRF_GNSS_USE_CASE_SINGLE_COLD_START | NRF_GNSS_USE_CASE_NORMAL_ACCURACY;

or

config.use_case = NRF_GNSS_USE_CASE_SINGLE_COLD_START | NRF_GNSS_USE_CASE_LOW_ACCURACY;

which would make the lines self documenting. I was not thinking about the use case where the application would read back the configuration and test it. I am also not sure how likely it is, but for it is possible and then these defines would not work.